test/core/collections.test.js

```diff
@@ -308,4 +308,30 @@ describe("Collections", () => {
 			expect(mockGmi.setGameData).not.toHaveBeenCalled();
 		});
 	});
+
+	describe("Returned setUnique method", () => {
+		test("Sets value on one item in collection", () => {
+			const collection = initCollection(mockScreen)("testCollection");
+			collection.setUnique({ id: "id1", key: "selected", value: true });
+
+			expect(mockGmi.setGameData.mock.calls[0][1].collections.testCollection[0]).toEqual({
+				id: "id1",
+				selected: true,
+			});
+			expect(collection.getAll()[0].selected).toBe(true);
```

Comment:
nitpick but you could assert that the other item doesn't have 'selected' on it, for added robustness